packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/mcp/formatters/intelligence.py

```
@@ -404,3 +417,64 @@ def format_document_clusters(clusters: dict[str, Any]) -> str:
+            # -------------------------
+            for r in path_edges:
+                edges.append(
+                    {
+                        "source": safe_get(r.src_node, "id"),
+                        "target": safe_get(r.dest_node, "id"),
+                        "type": r.type,
+                        "properties": (
+                            dict(r.properties) if hasattr(r, "properties") else {}
+                        ),
+                    }
+                )
```

Ruff (0.15.14)

[warning] Use `list.extend` to create a transformed list

(PERF401)